### PR TITLE
gc: only run compaction filter if cluster version is ok

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### What problem does this PR solve?
TiKV can't enable compaction filter during rolling update.

### What is changed and how it works?
Only enable compaction filter after cluster-version reaches specified point.

### Check List 

Tests 
- Unit test

Side effects

### Release note
- No release note

### CODE REVIEW

File: src/server/gc_worker/config.rs [hunk 1]
```diff
@@ -18,6 +23,7 @@ pub struct GcConfig {
     pub batch_keys: usize,
     pub max_write_bytes_per_sec: ReadableSize,
     pub enable_compaction_filter: bool,
+    pub compaction_filter_minimal_version: String,
```
Comment (Little-Wallace) [inline src/server/gc_worker/config.rs:26]: Why is it a configure variable rather than a const variable?

Comment (hicqu) [reply src/server/gc_worker/config.rs:26]: Some users can enable it before 5.0.0, if they will never fallback from 4.1 to 4.0.

Comment (Little-Wallace) [reply src/server/gc_worker/config.rs:26]: I thought we would never cherry-pick this PR to release-4.0 branch....
If we cherry-pick it to release-4.0, all users of 4.x will run GC-worker without raft-replication.

Comment (MyonKeminta) [reply src/server/gc_worker/config.rs:26]: At least I don't think it should be able to be configured in the config file. If we do need to cherry pick to 4.0 some day, we can submit a PR to update the constant.

File: src/server/gc_worker/compaction_filter.rs [hunk 2]
```diff
@@ -19,6 +19,11 @@ use txn_types::{Key, WriteRef, WriteType};
 const DEFAULT_DELETE_BATCH_SIZE: usize = 256 * 1024;
 const DEFAULT_DELETE_BATCH_COUNT: usize = 128;
 
+// The default version that can enable compaction filter for GC. This is necessary because after
+// compaction filter is enabled, it's impossible to fallback to ealier version which modifications
+// of GC are distributed to other replicas b Raft.
```
Comment (MyonKeminta) [inline src/server/gc_worker/compaction_filter.rs:24]:
```suggestion
// of GC are distributed to other replicas by Raft.
```

**Pull request merged**